package/test/beam_location_test.dart

```diff
@@ -135,4 +135,48 @@ void main() {
     delegate.beamToNamed('/');
     expect(delegate.currentBeamLocation, isA<NotFound>());
   });
+
+  group('pattern matching', () {
+    test('pattern matching works correctly when choosing routes', () {
+      void checkMatch(
+        String routeMatcher,
+        String routeToBeMatched,
+        bool shouldMatch,
+      ) {
+        expect(
+          RoutesBeamLocation.chooseRoutes(
+            RouteInformation(location: routeToBeMatched),
+            [routeMatcher],
+          ),
+          shouldMatch ? isNotEmpty : isEmpty,
+        );
+      }
+
+      checkMatch('/home', '/home', true);
+      checkMatch('/*', '/home', true);
+      checkMatch('/home/*', '/home/', true);
+      checkMatch('/home/*', '/home/one', true);
+      checkMatch('/home/*', '/home/one/two', true);
+      checkMatch('*', '/home', true);
+      checkMatch('*', '/home/one/two', true);
+      checkMatch('*', 'pretty much anything', true);
+      checkMatch('/*', '/home', true);
+      checkMatch('/*', 'home', true);
+      checkMatch('/*', '/home/one/two', true);
+      checkMatch('/*', 'home/one/two', true);
+      checkMatch('/*', '/leading slashes should be ignored', true);
+      checkMatch('/*', 'leading slashed should be ignored', true);
+
+      checkMatch('/', '/home', false);
+      checkMatch('/home', '/home/', false);
+      checkMatch('/home', '/home/one', false);
+      checkMatch('/home', '/home/one/two', false);
+      checkMatch('/home/*', '/home', false);
+      checkMatch('/home*', '/home/one', false);
+      checkMatch('*', '', false);
+      checkMatch('*', '/', false);
```

Comment:
I think these 2 should match, but I get into problems with other tests if I make them match.
I'll get back to it tomorrow and probably push some additional stuff into this PR.

You're right about the fix you provided, I just need to revisit how it affects other (maybe bad) tests.